core/meta/src/TClass.cxx

```diff
@@ -5464,29 +5464,24 @@ TClass *TClass::LoadClassCustom(const char *requestedname, Bool_t silent)
 
 void TClass::LoadClassInfo() const
 {
-   R__LOCKGUARD(gInterpreterMutex);
+   // Load class info if not already loaded and auto-parsing is not suspended
+   if (!fClassInfo && !gInterpreter->IsAutoParsingSuspended()) {
```

Comment:
In the end, if fClassInfo is made atomic, the start of the function can be

```
if (fClassInfo == nullptr && fCanLoadClassInfo) { ....
```
or
```
if (fClassInfo != nullptr || !fCanLoadClassInfo) return;
```
I am not sure which one is more efficient ...
You may even want to use R__likely/R__unlikely

```
if (R__unlikely(fClassInfo == nullptr && fCanLoadClassInfo)) { ....
// or
if (R__likely(fClassInfo != nullptr || !fCanLoadClassInfo)) return;
```

Cheers,
Philippe.